app/internal/statistics.py

```diff
@@ -0,0 +1,391 @@
+    """
+    num_of_days_in_period = (end - start).days + 1
+    if len(events_by_date) > 0:
+        min_events_in_day = min(day[1] for day in events_by_date)
+        max_events_in_day = max(day[1] for day in events_by_date)
+        sum_events_per_period = sum(day[1] for day in events_by_date)
+    else:
+        min_events_in_day = 0
+        max_events_in_day = 0
+        sum_events_per_period = 0
+    if num_of_days_in_period > len(events_by_date):
+        min_events_in_day = 0
+    avg_events_per_day = round(
+        sum_events_per_period / num_of_days_in_period, 2)
+    return DailyEventsStatistics(
+        min_events_in_day=min_events_in_day,
+        max_events_in_day=max_events_in_day,
+        avg_events_per_day=avg_events_per_day)
+
+
+def get_daily_events_statistics(
+        db: Session, userid: int, start: datetime.datetime,
+        end: datetime.datetime) -> Dict[str, Dict[str, int]]:
+    """ calculate statistics for daily events relevant for the requested user
+        and the requested date range. logic is performed in:
+        the db (get_events_by_date function),
+        while the rest is in the code (calc_daily_events_statistics function).
+
+    Args:
+        db: db session.
+        userid: requested user id number for statistics.
+        start: start of date range.
+        end: end of date range.
+
+    Returns: json with
+        day_events_stats: (for requested date range).
+            min_events_in_day: minimum number of daily events the user has.
+            max_events_in_day: maximum number of daily events the user has.
+            avg_events_in_day: average number of daily events the user has.
+    """
+    events_by_date = get_events_by_date(db, userid, start, end)
+    daily_events_statistics = calc_daily_events_statistics(
+        events_by_date, start, end)
+    return {
+        "day_events_stats": {
+            "min_events_in_day": daily_events_statistics.min_events_in_day,
+            "max_events_in_day": daily_events_statistics.max_events_in_day,
+            "avg_events_per_day": daily_events_statistics.avg_events_per_day}}
+
+
+def get_events_duration_statistics_from_db(
+        db: Session, userid: int, start: datetime.datetime,
+        end: datetime.datetime) -> NamedTuple('EventsDurationStatistics'):
+    """ get data of shortest, longest and average event duration from the db
+
+    Args:
+        db: db session.
+        userid: requested user id number for statistics.
+        start: start of date range.
+        end: end of date range.
+
+    Returns:
+        NamedTuple of: shortest_event, longest_event, average_event
+    """
+    events_duration_statistics = db.query(
+        (func.min(func.julianday(Event.end) - func.julianday(Event.start))
+         * NIN_IN_DAY),
+        (func.max(func.julianday(Event.end) - func.julianday(Event.start))
+         * NIN_IN_DAY),
+        (func.avg(func.julianday(Event.end) - func.julianday(Event.start))
+         * NIN_IN_DAY)
```

Comment:
Save `func.julianday(Event.end) - func.julianday(Event.start)` on a variable outside the `db.query`.  It will help this lines be much more readable